docs/core/classes.md

```diff
@@ -345,22 +308,20 @@ qx.Class.define("qx.test.Cat", {
   },
   members: {
     makeSound : function(x) {
-      this.constructor.someStaticMethod(x);
+      qx.test.Animal.someStaticMethod(x);
     }
   }
 });
 ```
 
 The syntax for accessing static variables simply is
-`this.constructor.someStaticVar`. Please note, for `this.constructor` to be
-available, the class must be a derived class of `qx.core.Object` , which is
-usually the case for regular, non-static classes.
-
-Instead of `this.constructor` you can also use the alternative syntax
-`this.self(arguments)`.
+`qx.test.Animal.someStaticVar`.  Please note that `this.constructor` is not
+recommended because it always refers to the subclass of the object, and therefore
+the implementation can change unexpectedly at run time.  Always specify class names
+explicitly in static methods.
 
-In purely static classes for calling a static method from another static method,
-you can directly use the `this` keyword, e.g. `this.someStaticMethod(x)`.
+You should also not use the `this` keyword in purely static classes, because static
```

Comment:
s/You should also not/Unless you know how the static method is called in all cases, you should also not/